.github/workflows/build.yml

```diff
@@ -0,0 +1,75 @@
+name: build and test
+on:
+  push:
+    branches-ignore:
+      - "gh-pages"
+    tags-ignore:
+      - "v0.[0-9]**"
+  pull_request:
+    branches-ignore:
+      - "gh-pages"
+
+jobs:
+  build_and_test:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+
+          # GCC builds
+          - CC: gcc
+            CONF_FLAGS:
+            DISTCHECK: 1
+            name_extra: Basic build and dist check
+
+          - CC: gcc
+            CONF_FLAGS: --without-simd
+            name_extra: Without simd
+
+          - CC: gcc
+            CONF_FLAGS: --host=i686-linux
+            CFLAGS: -m32
+            arch: i386
+            name_extra: 32-bit build
+
+          # clang builds
+          - CC: clang
+            CONF_FLAGS:
+            DISTCHECK: 1
+            name_extra: Basic build and dist check
+
+          - CC: clang
+            CONF_FLAGS: --without-simd
+            name_extra: Without simd
+
+          - CC: clang
+            CONF_FLAGS: --host=i686-linux
+            CFLAGS: -m32
+            arch: i386
+            name_extra: 32-bit build
+
+    name: ${{ matrix.name_extra }} (${{ matrix.CC }})
+    runs-on: ubuntu-latest
+    env:
+      CC: ${{ matrix.CC }}
+    steps:
+      - name: Define feature and arch dependent environment variables
+        # Note: any "variable=value" written to the $GITHUB_ENV file will be
+        # defined as an environment variable for all future steps in this job
+        # See: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable
+        run: |
+            exec >> $GITHUB_ENV
+            echo "XRDP_CFLAGS=-I$GITHUB_WORKSPACE/xrdp/common"
+            if [ -n "${{ matrix.CFLAGS }}" ]; then
+            echo "CFLAGS=${{ matrix.CFLAGS }}"
+            fi
```

Comment:
To me using `exec` is a non-obvious way to set the environment variables.

Reply:
Fair point.

This is a shell-ism to append all future output from the script to $GITHUB_ENV.

As it happens, after a good night's sleep, I was able to remove this run statement completely and move everything to the env variable above. I think this is a better approach. Do you agree?

Reply:
yeah, I think this is easier to understand.